### ISSUE TITLE
This Library isn't Compatible with Windows

### ISSUE THREAD

**Issue by MilesLitteral** [tmcdonell/cuda#70]
I don't know how else to put it but this library isn't Windows 10 compatible, there should be a note in the readme about this but using nm and ld on Windows is just going to give you nightmares like it's given me and even with hardcoding you can't build this library, please put a warning as to the fact this library is built in a Linux Environment, for Linux, and tested in Linux where, therefore, using it on Windows it's an uphill battle, No comment on Mac compatibility.

**Comment by tmcdonell** [comment]
`ld` and `nm` it should find from your mingw install (which ghc is using). Do the notes in the windows read me help at all, or are they just completely outdated now?

You are right, I don’t regularly test on windows, so I rely on people sending feedback/patches when things break.

**Comment by MilesLitteral** [comment]
Yes, I believe the instructions are currently outdated, I've made hotfixes which make the library build on Windows, I can open a pull request if you'd like to maybe address the issue fully? 

Oh Mingw, hmmm my Haskell build on Win64 is using Msys but I doubt thats the full issue??? All in all it will take some ground work which I don't mind helping out with at all

**Comment by tmcdonell** [comment]
If you do get it working yes please open a PR, that would be fantastic!

Windows support was added a long time ago via a contributor, but I believe they no longer need it thus it hasn't been kept up to date ):

**Comment by MilesLitteral** [comment]
Hey @tmcdonell  I have made progress with the Library and can report the following:

The issue seems two fold: 
First, the function ```defaultCUDAInstallPath``` (in Setup.exe) was acting wonky, in my branch I extrapolated the function to have platform specific behavior (it works very well)

Second ```cuda-0.11.0.0-inplace.conf``` gives the wrong path for the dlls:
library-dirs:
    ...\cuda-master\dist-newstyle\build\x86_64-windows\ghc-8.10.7\cuda-0.11.0.0\build
    C:\NVIDIA\Toolkit\CUDA\v9.0\lib\x64 C:\ghcup\msys64\mingw64\lib

dynamic-library-dirs:
    ...\cuda-master\dist-newstyle\build\x86_64-windows\ghc-8.10.7\cuda-0.11.0.0\build
    C:\NVIDIA\Toolkit\CUDA\v9.0\lib\x64 C:\ghcup\msys64\mingw64\lib

it seems to think the .lib and .dll files are in the same place based on the inplace, when you put the .dll it's looking for in the same place as the lib path and update the defautCUDAInstallPath function this library builds again

I will open a pull request soon, if you know what function is setting  ```dynamic-library-dirs``` I'd appreciate the pointer :)
